### PR TITLE
[M] Migrate rest of the pool_resource_spec to Java (ENT-5285)

### PR DESCRIPTION
**Pull request by bakajstep**
- Migrate remaining two tests
- Remove pool_resource_spec.rb

### CODE REVIEW

File: spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java [hunk 1]
```diff
@@ -565,4 +571,61 @@ public void shouldIncludeProductBrandingWhenFetchingPool() {
         assertTrue(b2filter.isPresent());
     }
 
+    @Test
+    @OnlyInStandalone
+    @DisplayName("should includes null values in json in Standalone")
+    public void shouldIncludesNullValuesInJsonInStandalone() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+        ApiClient ownerClient = this.createUserClient(adminClient, owner);
```
Comment (Ceiu) [inline spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java:580]: The owner client is extraneous here -- we're not interested in how the fetch is scoped.

File: spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java [hunk 2]
```diff
@@ -565,4 +571,61 @@ public void shouldIncludeProductBrandingWhenFetchingPool() {
...
+
+        ProductDTO product = Products.randomSKU();
+
+        adminClient.ownerProducts().createProductByOwner(owner.getKey(), product);
```
Comment (Ceiu) [inline spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java:584]: These two lines can be combined into one by using the in-class helper function for creating a product:

```
ProductDTO product = this.createProduct(adminClient, owner);
```

**Review by Ceiu — CHANGES_REQUESTED** [review]

**Review by Ceiu — APPROVED** [review]
We should get a second reviewer on this, given how much of it is shared.

File: spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java [hunk 3]
```diff
@@ -462,69 +460,64 @@ public void shouldIncludeCalculatedAttributesWhenFetchingPools() {
...
+        HostedTestApi upstreamClient = adminClient.hosted();
 
-    //     CdnDTO cdn = new CdnDTO()
-    //         .name(StringUtil.randomSuffix("test_cdn"))
-    //         .label("Test CDN")
-    //         .url("https://cdn.test.com");
+        CdnDTO cdn = new CdnDTO()
+            .name(StringUtil.random("test_cdn"))
+            .label(StringUtil.random("Test CDN"))
+            .url("https://cdn.test.com");
```
Comment (JoshAlbrecht23) [inline spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java:473]: In PR-3509 I added a Cdn builder and you can copy it over and make this a one liner with `CdnDTO cdn = Cdns.random()`.


https://github.com/candlepin/candlepin/pull/3509/files#diff-2028afa08357f43806ba504c152be9d7e173882e769f29f9b1ab9e65557c7970

File: spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java [hunk 4]
```diff
@@ -462,69 +460,64 @@ public void shouldIncludeCalculatedAttributesWhenFetchingPools() {
...
 
-    //     PoolDTO pool = Pools.randomUpstream(product)
-    //         .setCdn(cdn);
+        upstreamClient.createProduct(product);
+        upstreamClient.createSubscription(Subscriptions.random(owner, product).cdn(cdn));
 
-    //     pool = adminClient.owners()
-    //         .createPool(owner.getKey(), pool);
+        AsyncJobStatusDTO job =  adminClient.owners().refreshPools(owner.getKey(), false);
+        adminClient.jobs().waitForJob(job.getId());
```
Comment (JoshAlbrecht23) [inline spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java:487]: I would verify that the refresh pools jobs was successful by asserting the AsyncJobStatusDTO state equals "FINISHED".

File: spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java [hunk 5]
```diff
@@ -462,69 +460,64 @@ public void shouldIncludeCalculatedAttributesWhenFetchingPools() {
...
 
 
-    //     // We should have two pools created, but only the base/primary pool should have the CDN
-    //     // associated with it
-    //     List<PoolDTO> pools = adminClient.pools()
-    //         .listPools(owner.getId(), null, null, true, null);
+        // We should have two pools created, but only the base/primary pool should have the CDN
+        // associated with it
+        List<PoolDTO> pools = adminClient.pools()
+            .listPools(owner.getId(), null, null, true, null, null, null, null, null);
```
Comment (JoshAlbrecht23) [inline spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java:493]: You can use the `adminClient.pools().listPoolsByOwner(owner.getId())` method here.

**Review by JoshAlbrecht23 — CHANGES_REQUESTED** [review]
Some small changes

File: spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java (hunk 3 above)
Comment (bakajstep) [reply spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java]: done

File: spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java (hunk 4 above)
Comment (bakajstep) [reply spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java]: done

File: spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java (hunk 5 above)
Comment (bakajstep) [reply spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java]: done

**Review by JoshAlbrecht23 — APPROVED** [review]

**Pull request merged**